src/Features/Core/Portable/Workspace/BackgroundParser.cs

```diff
@@ -153,6 +161,12 @@ public void Parse(Document document)
         {
             if (document != null)
             {
+                if (SolutionCrawlerOptions.GetBackgroundAnalysisScope(document.Project) == BackgroundAnalysisScope.ActiveFile &&
+                    _documentTrackingService?.TryGetActiveDocument() != document.Id)
+                {
+                    return;
+                }
```

Comment:
1. same comments.  all this needs lots of documentation as to why this is correct.   how we are ok once user options change, etc. etc.
2. note the inconsistency here as well. you do this *outside* of the lock, and *before* the cancel operation below.  I basically cannot reason about these code changes and why they're so different and why that's what we want.

Reply:
Thanks, I have made both the documentation and implementation consistent in BackgroundParser and BackgroundCompiler now.